### PR TITLE
undervolt: init at 0.2.5

### PR DESCRIPTION
**Pull request by eadwu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by fpletz** [comment]
@GrahamcOfBorg build undervolt

File: nixos/modules/services/hardware/undervolt.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          ];
+          serviceConfig = {
+            ExecStart = ''
+              ${pkgs.undervolt}/bin/undervolt --core ${toString cfg.core} --cache ${toString cfg.cache} --analogio ${toString cfg.analogio} --uncore ${toString cfg.uncore} --gpu ${toString cfg.gpu}
+            '';
+          };
+          wantedBy = [
+            "suspend.target"
+            "hibernate.target"
+            "hybrid-sleep.target"
```
Comment (fpletz) [inline nixos/modules/services/hardware/undervolt.nix]: Did you test this? Aren't those three targets activated before the actual suspend/sleep?

The `powerManagement.resumeCommands` option is probably more useful.

Comment (eadwu) [reply nixos/modules/services/hardware/undervolt.nix]: Moved cfg under `powerManagement` while also using `powerManagement.powerUpCommands`.

Comment (eadwu) [reply nixos/modules/services/hardware/undervolt.nix]: Nevermind, `powerManagement.powerUpCommands` fails when booting up while `powerManagement.resumeCommands` doesn't run at boot. Just returned back to the service.

**Pull request closed without merging**